### PR TITLE
Add gh-action for npm publish on release create

### LINKED ISSUE #3: Add npm publishing to the CI post-merge hook
Depends on #2

### PR DESCRIPTION
**Pull request by danfinlay**
Related to #31

Pulled directly off [the github actions example for this](https://docs.github.com/en/actions/guides/publishing-nodejs-packages).

[Additional notes on configuration on team Roam](https://roamresearch.com/#/app/MetaMask/page/t4mYLgOZK)

### CODE REVIEW

**Review by kumavis — APPROVED** [review]

File: .github/workflows/npm-publish.yaml [hunk 1]
```diff
@@ -0,0 +1,19 @@
+name: Node.js Package
+on:
+  release:
+    types: [created]
+jobs:
+  build:
```
Comment (rekmarks) [inline .github/workflows/npm-publish.yaml:6]: Is the this the name of the job? Should it be `build`?

**Review by rekmarks — CHANGES_REQUESTED** [review]
Pending discussion / further review, probably during tooling week.

**Comment by rekmarks** [comment]
Discussion-to: #31

**Comment by rekmarks** [comment]
After working with GitHub Actions since April, I've become convinced that safely setting up npm publishing in CI will require someone's dedicated time and attention, likely over the course of weeks. That someone will probably be our future tooling engineer. Until then, I will close this PR. We can use it as a point of reference, if necessary.

**Pull request closed without merging**